inference-engine/src/cldnn_engine/ops/mvn.cpp

```diff
@@ -1,38 +1,83 @@
+// Copyright (C) 2020-2021 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
 
 #include "cldnn_program.h"
 #include "cldnn_common_utils.h"
 
 #include "ngraph/op/mvn.hpp"
+#include "ngraph/op/constant.hpp"
 
 #include "api/mvn.hpp"
+#include <algorithm>
 
 namespace CLDNNPlugin {
 
+bool checkAxesSuitability(const std::vector<int32_t>& axes, size_t shape_size) {
+    if (shape_size > 6 || (shape_size != axes.size() + 1 && shape_size != axes.size() + 2))
+        return false;
+    int value = shape_size - 1;
+    for (int i = axes.size() - 1; i >= 0; i--, value--) {
+        if (axes[i] != value)
+            return false;
+    }
+    return true;
+}
+
 void CreateMVNOp(Program& p, const std::shared_ptr<ngraph::op::v0::MVN>& op) {
     p.ValidateInputs(op, {1});
     auto inputPrimitives = p.GetInputPrimitiveIDs(op);
     std::string layerName = layer_type_name_ID(op);
 
+    bool across_channels = op->get_across_channels();
+    bool normalize_variance = op->get_normalize_variance();
+    float eps = op->get_eps();
+    bool eps_inside_sqrt = true;
+
+    auto mvnPrim = cldnn::mvn(layerName,
+                              inputPrimitives[0],
+                              normalize_variance,
+                              eps,
+                              eps_inside_sqrt,
+                              across_channels);
+
+    p.AddPrimitive(mvnPrim);
+    p.AddPrimitiveToProfiler(op);
+}
+
+void CreateMVNOp(Program& p, const std::shared_ptr<ngraph::op::v6::MVN>& op) {
+    p.ValidateInputs(op, {2});
+    auto inputPrimitives = p.GetInputPrimitiveIDs(op);
+    std::string layerName = layer_type_name_ID(op);
+
+    auto inConst = std::dynamic_pointer_cast<ngraph::op::Constant>(op->get_input_node_shared_ptr(1));
+    if (!inConst)
+        THROW_IE_EXCEPTION << "Unsupported parameter nodes type in " << op->get_friendly_name() << " (" << op->get_type_name() << ")";
+
+    std::vector<int32_t> axes = inConst->cast_vector<int32_t>();
+
+    if (!checkAxesSuitability(axes, op->get_input_shape(0).size())) {
+        THROW_IE_EXCEPTION << "Incorrect input 'axes' in " << op->get_friendly_name() << " (" << op->get_type_name() << ")";
+    }
+
     const size_t chanelAxis = 1;
-    ngraph::AxisSet reductionAxes = op->get_reduction_axes();
-    // FIXME: op->get_across_channels(); doesn't work for some reason. Is it expected?
-    bool across_channels = reductionAxes.count(chanelAxis) > 0;
+    bool across_channels = std::find(axes.begin(), axes.end(), chanelAxis) != axes.end();
     bool normalize_variance = op->get_normalize_variance();
     float eps = op->get_eps();
+    bool eps_inside_sqrt = op->get_eps_mode() == ngraph::op::MVNEpsMode::INSIDE_SQRT;
 
     auto mvnPrim = cldnn::mvn(layerName,
                               inputPrimitives[0],
-                              across_channels,
                               normalize_variance,
-                              eps);
+                              eps,
```

Comment:
There is a code duplication between MVN-0 and MVN-6. So I think most of the code should be extracted to a common function.